ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -1386,11 +1390,33 @@ private static void addObjectToKsqlArray(final KsqlArray array, final Object val
       array.add(SqlTimeTypes.formatDate((Date) value));
     } else if (value instanceof Time) {
       array.add(SqlTimeTypes.formatTime((Time) value));
+    } else if (value instanceof byte[]) {
+      array.add(serializeVertX3CompatibleByte((byte[]) value));
     } else {
       array.add(value);
     }
   }
 
+  /**
+   * VertX 4 changed to serialize using Base64 without padding but our server
+   * still encodes the data with padding - this uses the same serializer that
+   * the server uses to serialize the data into a string. Note that this issue
+   * only affects tests because the test uses string comparison to compare the
+   * bytes (it's schemaless, so the byte[] comes back as a string from the server).
+   *
+   * @see <a href=https://github.com/eclipse-vertx/vert.x/pull/3197>vertx#3197</a>
+   */
+  private static String serializeVertX3CompatibleByte(final byte[] bytes) {
```

Comment:
Will clients using VertX 3 be able to query ksqlDB running VertX 4?  (I suppose that's the goal here, right?)

Reply:
yup! that's the goal - our whole system will work in "vertx3 compatibilty mode"

Reply:
yup! see the inline comment:
```
   * Note that this issue
   * only affects tests because the test uses string comparison to compare the
   * bytes (it's schemaless, so the byte[] comes back as a string from the server).
```